src/main/kotlin/org/rust/lang/core/resolve/RustResolveEngine.kt

```diff
@@ -159,6 +182,17 @@ object RustResolveEngine {
             else
                 current.parentOfType<RustResolveScope>()
         }
+
+    private fun Project.getPackage(name: String): CargoProjectDescription.Package? =
```

Comment:
This should not be a `Project` extension. In intellij, the global context for a "project" is actually a `Module`. You can have two completely different Cargo projects added as different intellij modules in the same IDEA project.

More specifically, the IDEA itslf allows you to create several `Modules` within a single project, while IDEs like PyCharm generally create only a single `Module` for a `Project`.

So, let's make this a method of `CargoProjectDescription` (which is properly attached to a module) and make it a caller's responsibility to retrieve the necessary module. The module is available in `CargoCommandConfiguration`: https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/cargo/runconfig/CargoCommandConfiguration.kt#L33, you'll need to thread it to the filter.